### PR TITLE
Bump baseline jenkins core to 2.222 for milestones / system read + cleanup

### PR DESCRIPTION
**Pull request by timja**
Bumps the baseline to 2.222

This pull request is probably best viewed commit by commit,
It removes unneeded code on newer versions,
and fixes deprecation warnings introduced by the core / plugin-pom bump

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Did you provide a test-case? That demonstrates feature works or fixes the issue.

### CODE REVIEW

File: plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
     private static final Logger LOGGER = Logger.getLogger(BaseConfigurator.class.getName());
 
-    @NonNull
+    @Nonnull
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: 🤔 I still think this applies: https://blog.codefx.org/java/jsr-305-java-9/#Alternatives-To-JSR-305

File: pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.53</version>
+    <version>4.0-beta-5</version>
```
Comment (casz) [inline pom.xml]: I would still like to see `-Duse-jenkins-bom` in the mvn config

Comment (timja) [reply pom.xml]: It’s on by default in the 4.x line afaik

File: plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java (hunk 1 above)
Comment (timja) [reply plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: As far as I can tell the old ones are deprecated and point to these unless I’m missing something?

File: pom.xml (hunk 2 above)
Comment (casz) [reply pom.xml]: neat I must have missed that 😅

File: plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java (hunk 1 above)
Comment (casz) [reply plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: NonNull should not be deprecated: https://github.com/spotbugs/spotbugs/issues/130

Comment (timja) [reply plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: K will look tomorrow at why we don’t have the spotbugs version

Comment (timja) [reply plugin/src/main/java/io/jenkins/plugins/casc/BaseConfigurator.java]: It's coming from lib-version-number in jenkins-core by the looks of it, its fixed on master but un released

https://github.com/jenkins-infra/repository-permissions-updater/pull/1434

Will revert the changes here, and depending on timing bump the versions in this PR so no deprecation introduced here, or do in a follow up

File: plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java [hunk 3]
```diff
@@ -31,15 +27,14 @@
...
-    private static final VersionNumber MIN_VERSION_FOR_HASHED_PASSWORDS = new VersionNumber("2.161");
     /// matches HudsonPrivateSecurityRealm.JBCRYPT_HEADER
     private static final String HASHED_PASSWORD_PREFIX = "#jbcrypt:";
 
     public HudsonPrivateSecurityRealmConfigurator() {
         super(HudsonPrivateSecurityRealm.class);
     }
 
-    @NonNull
+    @Nonnull
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java]: 😱

File: plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java [hunk 4]
```diff
@@ -1,24 +1,20 @@
...
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
 import java.util.Collection;
 import java.util.Set;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 import java.util.stream.Collectors;
-import jenkins.model.Jenkins;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java]: 😱

File: plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java (hunk 3 above)
Comment (timja) [reply plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java]: hmm dunno what happened there, fixing

**Comment by timja** [comment]
Added breaking for the proxy password being renamed

**Comment by timja** [comment]
Codacy is moaning about all the changes from `getInstance` to `get` don't have good coverage around them

**Pull request merged**